packages/sysutils/busybox/scripts/init

```diff
@@ -341,6 +341,68 @@
     $MOUNT_CMD "$MOUNT_TARGET" "$2" "$3" "$4"
   }
 
+  # mount the specified SYSTEM file and output arch from /etc/os-release
+  get_project_arch() {
+    mount_part "$1" "/sysroot" "ro,loop"
+
+    if [ -f /sysroot/etc/os-release ]; then
+      . /sysroot/etc/os-release
+      echo "${LIBREELEC_ARCH:-${OPENELEC_ARCH}}"
+    fi
+
+    umount /sysroot
+  }
+
+  # If the project/arch of current matches the upgrade, then it is considered compatible.
+  # Otherwise, mount the upgrade SYSTEM partition and, if canupdate.sh is available,
+  # call the script to determine if the current upgrade file can be applied on to the
+  # current system - 0 means it is compatible, non-zero that it is not compatible.
+  is_compatible() {
+    local result=1
+
+    if [ "${2}" = "${3}" ]; then
+      result=0
+    else
+      mount_part "$1" "/sysroot" "ro,loop"
+
+      if [ -f /sysroot/usr/share/bootloader/canupdate.sh ]; then
+        sh /sysroot/usr/share/bootloader/canupdate.sh "${2}" "${3}" && result=0
+      fi
+
+      umount /sysroot
+    fi
+
+    return ${result}
+  }
+
+  # determine if the new SYSTEM file is compatible with the current SYSTEM file
+  check_is_compatible() {
+    local update_filename="${1}"
+    local old_system="${2}"
+    local new_system="${3}"
+    local old_project_arch="$(get_project_arch "${old_system}")"
+    local new_project_arch="$(get_project_arch "${new_system}")"
+
+    # If old or new project/arch isn't available then could be very old (pre-/etc/os-release) build - have to trust it
+    if [ -n "${old_project_arch}" -a -n "${new_project_arch}" ]; then
+      # If the old project/arch is not compatible with the new project/arch then abort...
+      if ! is_compatible "${new_system}" "${old_project_arch}" "${new_project_arch}"; then
+        echo ""
+        echo "ERROR: $(basename "${update_filename}") is not compatible with ${old_project_arch} hardware - update cancelled."
+        echo ""
+        echo "Current system: ${old_project_arch}"
+        echo "Update  system: ${new_project_arch}"
+        echo ""
+        echo "If you know what you are doing, create \"$UPDATE_ROOT/.nocompat\" to avoid this check."
+        echo "However, doing so may result in an unbootable system."
+        echo ""
```

Comment:
"Create /storage/.update/.nocompat to disable compatibility checks and risk a non-booting system"

^^ less wordy one-liner for non-English users